ADAL/src/utils/NSString+ADHelperMethods.m

```diff
@@ -286,6 +286,27 @@ + (BOOL)adIsStringNilOrBlank:(NSString *)string
     return [string rangeOfCharacterFromSet:nonWhiteCharSet].location == NSNotFound;
 }
 
+- (long)adFindCharactersFromSet:(NSCharacterSet *)set
+                           start:(long)startIndex
+{
+    THROW_ON_NIL_ARGUMENT(set);
+    long end = self.length;
+    if (startIndex >= end)
+        return end;
+    
+    NSRange toSearch = {.location  = startIndex, .length = (end - startIndex)};
+    long found = [self rangeOfCharacterFromSet:set options:NSLiteralSearch range:toSearch].location;
+    return (found == NSNotFound) ? end : found;
```

Comment:
I pulled the previous implementation to fix this. What I understood is that the caller is doing substringWithRange, if we use NSNotFound, which is max long, call to substringWithRange will go out of range.
Ideally this should be generic and let the caller to handle this case. Since it is well tested before, I will leave it this way for now. Does that make sense?